pydash/objects.py

```diff
@@ -994,10 +996,30 @@ def omit_by(obj, iteratee=None):
     if not callable(iteratee):
         keys = iteratee if iteratee is not None else []
 
-        def iteratee(value, key):  # pylint: disable=function-redefined
-            return key in keys
+        if isinstance(keys, list):
+            paths = [to_path(key) for key in keys]
+
+            grouped_paths = pyd.group_lists(paths)
+        else:
+            grouped_paths = keys
+
+        ret = {}
+        for key, value in iterator(obj):
+
+            if key in grouped_paths and not grouped_paths[key]:
```

Comment:
Are you suggesting that I simply implement this as the omit_by function or do you think there is more that is needed.  That method does seem much cleaner that my way.

Reply:
I think my suggestion should do the trick. Could add a check on the path keys to determine if it's a deep path and switch between `pyd.clone` and `pyd.deep_clone` since deep-clone is more intensive and only needed if un-setting a deep path.

Reply:
Correction: I think it'd need to be something like `pydash.map_(iteratee, to_path)` instead of just `to_path(iteratee)` since `to_path` expects a single path string.

Reply:
I ended up modifying your suggestion a little bit as this method didn't work when the object is a list or when the iteratee is a function.  If the object is a list I simply convert it to a dict and if the iteratee is a function, I use the old method.